lib/connection.js

```diff
@@ -1388,6 +1388,21 @@ Connection.prototype.syncIndexes = async function syncIndexes() {
  * Switches to a different database using the same connection pool.
  *
  * Returns a new connection object, with the new db.
+ * 
+ * ####Example
+ *    async function createMultiTenatConnection(tenatId, modelName, schema) {
+  *     const conn = await mongoose.createConnection('mongodb://localhost:27017/test');
+  *     const dbName = `db_${tenatId}`;
+  *     const dbOptions = {
+    *     //If true, create one connection object with multiple calls to useDb(). Default id false
+    *     userCache:true 
+    *     //If true, db will not listen to the events on original connection. Default is false
+    *     noListener:true
+  *     }
+  *     const db = mongoose.connection.useDb(dbName, dbOptions);
+  *     db.model(modelname, schema);
```

Comment:
Need to be careful with `db.model()` and `useCache`. This line may throw an error if reusing an existing connection. Add a check:

```javascript
if (db.models[modelname] == null) {
  db.model(modelname, schema);
}
```

Reply:
I'll make necessary correction for that.